doc/src/manual/modules.md

```diff
@@ -168,6 +168,23 @@ Importantly, the module name `NiceStuff` will *not* be in the namespace. If you
 julia> using .NiceStuff: nice, DOG, NiceStuff
 ```
 
+!!! note
+    Qualifying the names being used as in `using NiceStuff: NiceStuff, nice` is recommended over plain
+    `using Foo` for released packages, and other code which is meant to be re-used in the future with
+    updated dependencies or future versions of julia.
+
+    The reason for this is if another dependency starts to export one of the
+    same names as `Foo` the code will error due to an ambiguity in which
+    package the name should be taken from. This is especially problematic in
+    released packages which needs to be forward-compatible with the future
+    releases (e.g., when `Foo = "1"` and `Bar = 2` are listed in `[deps]`
+    section of the `Project.toml` of your package, it should be compatible
+    with the future releases of Foo 1.x and Bar 2.y which both export a
+    function `baz`).
```

Comment:
If I didn't already know what this is about, I am not sure I would understand. I think the point you are trying to make is: suppose you are doing `using Foo, Bar`, and it works. But then `Foo` 1.1 exports a new symbol `baz`, while `Bar` already exports a different `baz`. Then suddenly `using Foo, Bar` fails.